emulator/periph/src/flash_ctrl.rs

```diff
@@ -58,50 +57,21 @@ pub enum FlashOpError {
     InvalidOp = 3,
 }
 
-// Define a struct to represent the flash controller peripheral emulation.
-// Includes IRQs, timers, and peripheral registers, backend storage and a buffer page_size for the flash controller.
-#[derive(Bus)]
+// Define a dummy flash controller peripheral.
 pub struct DummyFlashCtrl {
-    #[register(offset = 0x00)]
-    interrupt_state: ReadWriteRegister<u32, InterruptState::Register>,
-
-    #[register(offset = 0x04)]
-    interrupt_enable: ReadWriteRegister<u32, InterruptEnable::Register>,
-
-    #[register(offset = 0x08)]
+    interrupt_state: ReadWriteRegister<u32, FlInterruptState::Register>,
+    interrupt_enable: ReadWriteRegister<u32, FlInterruptEnable::Register>,
     page_size: ReadWriteRegister<u32>,
-
-    #[register(offset = 0x0c)]
     page_num: ReadWriteRegister<u32>,
-
-    #[register(offset = 0x10)]
     page_addr: ReadWriteRegister<u32>,
-
-    #[register(offset = 0x14)]
-    control: ReadWriteRegister<u32, Control::Register>,
-
-    #[register(offset = 0x18)]
+    control: ReadWriteRegister<u32, FlControl::Register>,
     op_status: ReadWriteRegister<u32, OpStatus::Register>,
-
-    #[register(offset = 0x1c)]
     ctrl_regwen: ReadOnlyRegister<u32, CtrlRegwen::Register>,
-
-    /// Timer
     timer: Timer,
-
-    /// Backend flash storage file
     file: Option<File>,
-
-    /// Internal buffer for read, write operations
     buffer: Vec<u8>,
-
-    /// Operation pending flag
-    operation_pending: bool,
-
-    /// Error interrupt
+    operation_start: Option<ActionHandle>,
     _error_irq: Irq,
```

Comment:
Since we're using these, we can remove the underscore from the beginning. (The underscore is just to prevent the warnings when they are unused.)